plugins/inputs/pulsar_consumer/pulsar_consumer.go

```diff
@@ -0,0 +1,248 @@
+	DialTimeout string `toml:"dial_timeout,omitempty"`
+	RecvTimeout string `toml:"recv_timeout,omitempty"`
+	recvTimeout time.Duration
+
+	PingFrequency         string `toml:"ping_frequency,omitempty"`
+	PingTimeout           string `toml:"ping_timeout,omitempty"`
+	InitialReconnectDelay string `toml:"initial_reconnect_delay,omitempty"`
+	MaxReconnectDelay     string `toml:"max_reconnect_delay,omitempty"`
+	NewConsumerTimeout    string `toml:"new_consumer_timeout,omitempty"`
+
+	Topic     string `toml:"topic"`
+	Name      string `toml:"name"`
+	Exclusive bool   `toml:"exclusive,omitempty"`
+	QueueSize int    `toml:"queue_size,omitempty"`
+
+	MaxUndeliveredMessages int `toml:"max_undelivered_messages,omitempty"`
+}
+
+var sampleConfig = `
+  # URL to Pulsar cluster
+  # If you use SSL, then the protocol should be "pulsar+ssl"
+  url = "pulsar://localhost:6650"
+
+  # Timeout while trying to connect
+  dial_timeout = "15s"
+
+  # Timeout while trying to receive message
+  recv_timeout = "5s"
+
+  # Topic of message
+  topic = ""
+
+  # Name of the consumer
+  name = ""
+
+  ## Is consumer exclusive
+  # exclusive = false
+
+  ## Queue size for this consumer
+  # queue_size = 1000
+
+  ## Path to certificates and key for TLS
+  # tls_ca = ""
+  # tls_cert = ""
+  # tls_key = ""
+
+  ## Other optionals
+  ping_frequency = "1s"
+  ping_timeout = "1s"
+  initial_reconnect_delay = "3s"
+  max_reconnect_delay = "10s"
+  new_consumer_timeout = "10s"
+
+  ## Maximum messages to read from the broker that have not been written by an
+  ## output.  For best throughput set based on the number of metrics within
+  ## each message and the size of the output's metric_batch_size.
+  ##
+  ## For example, if each message from the queue contains 10 metrics and the
+  ## output metric_batch_size is 1000, setting this to 100 will ensure that a
+  ## full batch is collected and the write is triggered immediately without
+  ## waiting until the next flush_interval.
+  # max_undelivered_messages = 1000
+
+  ## Data format to consume.
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
+  data_format = "influx"
+`
+
+func (p *pulsarConsumer) SampleConfig() string {
+	return sampleConfig
+}
+
+func (p *pulsarConsumer) Description() string {
+	return "Read metrics from Pulsar topic"
+}
+
+func (p *pulsarConsumer) SetParser(parser parsers.Parser) {
+	p.parser = parser
+}
+
+func (p *pulsarConsumer) Start(acc telegraf.Accumulator) error {
+	p.acc = acc.WithTracking(p.MaxUndeliveredMessages)
+	ctx, cancel := context.WithCancel(context.Background())
+	p.cancel = cancel
+
+	if p.connPool == nil && p.consumer == nil {
+		p.in = make(chan pulsar.Message, p.MaxUndeliveredMessages)
+		p.errs = make(chan error)
+
+		var err error
+		p.recvTimeout, err = time.ParseDuration(p.RecvTimeout)
+
+		conf := pulsar.ManagedConsumerConfig{}
```

Comment:
Nitpick: After changing the types to `internal.Duration`, define what you can of the struct at the same time:
```go
conf := pulsar.ManagedConsumerConfig{
  Addr: p.URL,
  Errs: p.errs,
...
```